packages/system/package.json

```diff
@@ -9,15 +9,17 @@
   ],
   "homepage": "https://gympass.github.io/yoga/",
   "license": "MIT",
-  "main": "src",
+  "main": "./src",
+  "types": "./src",
   "repository": {
     "type": "git",
     "url": "git+https://github.com/Gympass/yoga.git"
   },
   "scripts": {
-    "build": "yarn build:cjs && yarn build:esm",
-    "build:cjs": "NODE_ENV=cjs babel ./src --out-dir dist/cjs",
-    "build:esm": "NODE_ENV=esm babel ./src --out-dir dist/esm",
+    "build": "yarn build:types && yarn build:cjs && yarn build:esm",
+    "build:types": "tsup src !src/*.test.js !src/*.test.ts --outDir dist/typings --dts-only",
```

Comment:
I think that the `tests` files could be removed through the `tsconfig` file.
Just create a `tsconfig.json` file in the root of the package and extend the one on the root of the project, and then use the `include` property to include the `src`  folder and the `exclude` to remove test files from the build.